types/mongoose-paginate-v2/index.d.ts

```diff
@@ -0,0 +1,68 @@
+// Type definitions for mongoose-paginate-v2 1.0.13
+// Project: https://github.com/edwardhotchkiss/mongoose-paginate
+// Definitions by: Linus Brolin <https://github.com/linusbrolin>
+//                 simonxca <https://github.com/simonxca>
+//                 woutgg <https://github.com/woutgg>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+//
+// Based on type declarations for mongoose-paginate 5.0.0.
+
+/// <reference types="mongoose" />
+
+declare module 'mongoose' {
+  export interface CustomLabels {
+    totalDocs?: string,
+    limit?: string,
+    page?: string,
+    totalPages?: string,
+    docs?: string,
+    nextPage?: string,
+    prevPage?: string,
+  }
+
+  export interface PaginateOptions {
+    select?: Object | string;
+    sort?: Object | string;
+    customLabels?: CustomLabels,
+    collation?: CollationOptions,
+    populate?: Array<Object> | Array<string> | Object | string;
+    lean?: boolean;
+    leanWithId?: boolean;
+    offset?: number;
+    page?: number;
+    limit?: number;
+  }
+
+  export interface PaginateResult<T> {
+    docs: Array<T>;
+    total: number;
+    limit: number;
+    page?: number;
+    pages?: number;
+    offset?: number;
+    [customLabel: string]: Array<T> | number | undefined;
+  }
+
+  interface PaginateModel<T extends Document> extends Model<T> {
+    paginate(query?: Object, options?: PaginateOptions, callback?: (err: any, result: PaginateResult<T>) => void): Promise<PaginateResult<T>>;
+  }
+
+  export function model<T extends Document>(
```

Comment:
The type parameters on these overloads can't be inferred, so it would be better to leave them off and require users to add a cast, just like the tests currently do. In other words, there should only be one signature, and it shouldn't have a type parameter:

```ts
export function model(
  name: string, 
  schema?: Schema, 
  collection?: string, 
  skipInit?: boolean): PaginateModel<any>
```

Reply:
Sorry for the delay, illness kept me afk.
This change should probably be backported to mongoose-paginate as well, right?